File: LeetCode/DPs/Day2/746_min_cost_climbing_stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        DP = [0] * len(cost)
        DP[0], DP[1] = cost[0], cost[1]
        for i in range(2, len(cost)):
            DP[i] = cost[i] + min(DP[i-1], DP[i-2])
        return min(DP[-1], DP[-2])

File: LeetCode/DayChallenges/DayChallenge_2022_July/746_Min_Cost_Climbing_Stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
 var minCostClimbingStairs = function(cost) {
  const dp = [cost[0], cost[1]];
  
  for (let i = 2; i < cost.length; i += 1) {
      dp.push(Math.min(dp[i-1], dp[i-2]) + cost[i]);
  }
  return Math.min(dp[dp.length-1], dp[dp.length-2]);
};